Mark gp_external_grant_privileges as deprecated

danielgustafsson opened the pull request:
The `gp_external_grant_privileges` GUC was needed before 4.0 to let non superusers create external tables for gphdfs and http protocols. This is now handled via the normal GRANT mechanism but the GUC has not yet been removed. This is not something we should carry forward, especially if we want to change inner plumbings of external tables over to FDW with future upstream merges. Mark as deprecated now for 5.0 so that we can remove it in 6.0. If approved it requires a corresponding doc patch (and perhaps an updated entry in the base config file where it's listed as default off).

@ivannovick @Eulerizeit

danielgustafsson commented:
Rebased on top of current master

danielgustafsson commented:
This PR now has two commits, one simply marking it as DEPRECATED to throw a warning and one completely removing it. Depending on what we decide, one of these should be applied for 5.0.

Eulerizeit commented:
@schubert Can you look at this as well.

schubert commented:
@ivannovick - same question - how do you want to handle this deprecation?

ivannovick commented:
We should totally remove it in 5.0.0 and rely on normal GRANT.  And we need to document the product change in the release notes @mkiyama @dyozie

mkiyama commented:
In the currrent 4.3.x docs, a deprecation notice is in the description of the guc.
According to my notes, the deprecation notice was set in 4.3.11.2

danielgustafsson commented:
Ok, PR updated so that the GUC is completely removed and all traces from the documentation has been purged with minor rewrites where needed. I've also added a placeholder release notes entry, not sure where it's supposed to go but to avoid forgetting I added something there.

danielgustafsson commented:
Pushed to master with a release notes note, thanks for review!
